NPM Install: react-scripts failing to install (all platforms)

<!--
Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with esy@latest. It may have already been fixed.  -->
**esy version:** `master`
**Operating System:**  OSX / Linux

**Issue:**
```
*** installing react-scripts
*** sandboxPath: /tmp/dcf42ef7a2dd316e
EXEC: /Users/travis/build/esy/esy/_build/default/esy/bin/esyCommand.exe install
info install 0.2.11
info resolving esy packages: done
info solving esy constraints: done
error: path constraints path:utils are not allowed from archive:https://registry.npmjs.org/eslint-plugin-import/-/eslint-plugin-import-2.14.0.tgz#sha1:6b17626d2e3e6ad52cfce8807a845d15e22111a8
  resolving request eslint-plugin-import@=2.14.0
  solving dependencies of react-scripts@2.0.4
esy: exiting due to errors above
child_process.js:645
    throw err;
    ^
Error: Command failed: /Users/travis/build/esy/esy/_build/default/esy/bin/esyCommand.exe install
    at checkExecSyncError (child_process.js:602:13)
    at Object.execSync (child_process.js:642:13)
    at exec (/Users/travis/build/esy/esy/test-e2e-slow/setup.js:93:18)
    at retry (/Users/travis/build/esy/esy/test-e2e-slow/setup.js:110:26)
    at retry (/Users/travis/build/esy/esy/test-e2e-slow/setup.js:63:16)
    at Object.esy (/Users/travis/build/esy/esy/test-e2e-slow/setup.js:110:14)
    at Object.<anonymous> (/Users/travis/build/esy/esy/test-e2e-slow/install-npm.test.js:77:11)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
make[1]: *** [test-e2e-slow] Error 1
make: *** [ci] Error 2
```
Full log: https://travis-ci.org/esy/esy/jobs/437734150

--- a/esyi/PackageJson.ml
+++ b/esyi/PackageJson.ml
@@ -9,7 +9,6 @@ module Manifest = struct
     name : string option [@default None];
     version : SemverVersion.Version.t option [@default None];
     dependencies : Package.NpmFormula.t [@default Package.NpmFormula.empty];
-    devDependencies : Package.NpmFormula.t [@default Package.NpmFormula.empty];
     esy : EsyPackageJson.t option [@default None];
     dist : dist option [@default None]
   } [@@deriving of_yojson { strict = false }]
@@ -24,6 +23,12 @@ end
 module ResolutionsOfManifest = struct
   type t = {
     resolutions : (Package.Resolutions.t [@default Package.Resolutions.empty]);
+  } [@@deriving of_yojson { strict = false }]
+end
+
+module DevDependenciesOfManifest = struct
+  type t = {
+    devDependencies : Package.NpmFormula.t [@default Package.NpmFormula.empty];
   } [@@deriving of_yojson { strict = false }]
 end
 
@@ -50,7 +55,13 @@ let rebaseDependencies source reqs =
   in
   Result.List.map ~f reqs
 
-let packageOfJson ?(parseResolutions=false) ?source ~name ~version json =
+let packageOfJson
+  ?(parseResolutions=false)
+  ?(parseDevDependencies=false)
+  ?source
+  ~name
+  ~version
+  json =
   let open Run.Syntax in
   let%bind pkgJson = Json.parseJsonWith Manifest.of_yojson json in
   let originalVersion =
@@ -82,7 +93,17 @@ let packageOfJson ?(parseResolutions=false) ?source ~name ~version json =
   in
 
   let%bind dependencies = rebaseDependencies source dependencies in
-  let%bind devDependencies = rebaseDependencies source pkgJson.devDependencies in
+
+  let%bind devDependencies =
+    match parseDevDependencies with
+    | false -> return Package.NpmFormula.empty
+    | true ->
+      let%bind {DevDependenciesOfManifest. devDependencies} =
+        Json.parseJsonWith DevDependenciesOfManifest.of_yojson json
+      in
+      let%bind devDependencies = rebaseDependencies source devDependencies in
+      return devDependencies
+  in
 
   let%bind resolutions =
     match parseResolutions with

--- a/esyi/PackageJson.mli
+++ b/esyi/PackageJson.mli
@@ -1,5 +1,6 @@
 val packageOfJson :
   ?parseResolutions:bool
+  -> ?parseDevDependencies:bool
   -> ?source:Source.t
   -> name:string
   -> version:Version.t

--- a/esyi/Resolver.ml
+++ b/esyi/Resolver.ml
@@ -175,6 +175,7 @@ let packageOfSource ~allowEmptyPackage ~name ~overrides (source : Source.t) reso
         let%bind json = Json.parse data in
         PackageJson.packageOfJson
           ~parseResolutions:true
+          ~parseDevDependencies:true
           ~name
           ~version:(Version.Source source)
           ~source

